tiledb/sm/array/array.h

```diff
@@ -641,6 +644,31 @@ class Array {
    */
   std::mutex mtx_;
 
+  /** struct used for gathering various max tile sizes. */
+  struct tiledb_fragment_max_tile_sizes_t {
```

Comment:
This `struct` would be better defined outside `class Array`.
* It's the proper return value from a per-fragment function that computes the sizes.
* It can be the argument type of a function `maxima(TileSizes x, TileSizes y)` that computes a per-element max.
  * Alternately `TileSizes::maximize(TileSizes x)`.
  * There's already a _de facto_ implementation of this in the code.
* `operator=` can be the default
* It seems unlikely that the current need for an API function to measure legacy data is going to be the last time we need to deal with array and fragment tile sizes.

The naming convention is the C API one; it would be better to shorten it for use inside the core.